ui/src/app/edge/live/flat/flat-widget-line/abstract-flat-widget-line.ts

```diff
@@ -38,19 +36,19 @@ export abstract class AbstractFlatWidgetLine implements OnDestroy {
     ) {
     }
 
-    protected setValue(value: any) {
-        this.displayValue = this.converter(value);
+    protected setValue(value: any, showNotWhenValueEquals0?: boolean) {
+        this.displayValue = this.converter(value, showNotWhenValueEquals0);
     }
 
-    protected subscribe(channelAddress: ChannelAddress) {
+    protected subscribe(channelAddress: ChannelAddress, showNotWhenValueEquals0?: boolean) {
```

Comment:
I kind of understand why you need this weird `showNotWhenValueEquals0`, but I don't like it too much.

It should really be solved by using two different static Converters - one that shows zeros and that other one doesn't. Like this you do not need to spill this logic all over the place.

Second, but less prefered option: if you really need it, I would always keep it 'positive' to avoid 'double negative' (doppelte Verneinung) because it makes it very hard to understand the actual logic. I like a bit more the `showWhen0` that you use elsewhere.

Reply:
I only implmented this input() param with 'double negative', because if called in html it becomes 'false' without giving value. But you're completely right, i didn't like it too. But to have 2 converters for nearly the same converter, is not effective. 
And if i think about all the converters, which will be implemented in the future, i wasn't sure. 

Anyway for this example i don't need this second converter by now

Reply:
Having two converters with nearly same logic is no problem in general. Especially if it avoids spoiling the other code, like in this example.

See the Modbus `ElementToChannelConverter` in OpenEMS Edge, where we use a similar technique. It is used to convert from a value read via Modbus interface to a Channel value, e.g. adjusting the value to the proper unit. https://github.com/OpenEMS/openems/blob/develop/io.openems.edge.bridge.modbus/src/io/openems/edge/bridge/modbus/api/ElementToChannelConverter.java